Improve some more documentation

flagbug opened the pull request.

flagbug committed:
Flesh out some explanations and move the ViewModel code to the front

flagbug committed:
Explain the return value of OneWayBind and friends and explain where we get the events observable from

flagbug committed:
ReactiveAsyncCommand -> ReactiveCommand

flagbug committed:
Write some words about the solution layout

shiftkey commented on docs/sources/getting-started/ReactiveUI-101.md:
```diff
@@ -26,90 +29,154 @@ public MainWindow()
+        // Finally, we use RxUI's InvokeCommand operator, which takes the String 
+        // and calls the Execute method on the ExecuteSearch Command, after 
+        // making sure the Command can be executed via calling CanExecute.
+        this.ObservableForProperty(x => x.SearchTerm)
```
I think we should :fire: any documented usages of `ObservableForProperty` in favour of `WhenAny`/ `WhenAnyValue`. Thoughts?

flagbug replied:
Yup good idea, I think I've never even used `ObservableForProperty`

ghuntley replied:
:shipit:

shiftkey commented on docs/sources/getting-started/ReactiveUI-101.md:
```diff
@@ -26,90 +29,154 @@ public MainWindow()
-                .Select(x => tagRegex.Replace(HttpUtility.HtmlDecode(x.Value), ""));
+    public AppViewModel()
+    {
+        ExecuteSearch = ReactiveCommand.CreateAsyncTask(parameter => GetSearchResultsFromFlickr(this.SearchTerm));
```
:lipstick: is it worth introducing a subscription on `ThrownExceptions` here so we can make this a bit more feature-complete?

flagbug replied:
Probably? I'll see what I can come up with!

ghuntley replied:
Agree w/@shiftkey - all examples of `ReactiveCommand` usage should have a ThrownExceptions  subscription, even if it does nothing just to hammer on home how it works.

flagbug replied:
I'll try to combine `ThrownExceptions` with a `UserError` handler, so that that's explained too, but it'll have to wait until tomorrow :smile:

shiftkey replied:
I wouldn't worry too much about cramming in _every_ concept here. I mostly wanted `ThrownExceptions` here so that the user is aware about how to handle errors (rather than doing it inside `GetSearchResultsFromFlickr`)

shiftkey commented on docs/sources/getting-started/ReactiveUI-101.md:
```diff
@@ -26,90 +29,154 @@ public MainWindow()
+        // ToProperty operator, which is a helper to create an 
+        // ObservableAsPropertyHelper object.
+
+        _SpinnerVisibility = ExecuteSearch.IsExecuting.Select(x=> x ? Visibility.Visible : Visibility.Collapsed)                
```
:lipstick: bring the `.Select(...` down onto the next line for some better readability

flagbug replied:
Yup! Will do that when I change the other things :+1:

shiftkey commented:
:heart: :heart: :heart: :heart: :heart:

A few little things, nothing too serious.

flagbug committed:
Shorten the code with C# 6

flagbug committed:
Replace ObservableForProperty with WhenAnyValue

To preserve the Select call, we now trim the whitespaces from the search
term, which makes sense anyway

flagbug committed:
Pull the Select call down, for better readability

flagbug committed:
Add a subscription to the "ThrownExceptions" of the ReactiveCommand

flagbug commented:
@ghuntley @shiftkey Made the changes based on your feedback, thanks!

shiftkey commented:
LGTM - I'll let @ghuntley have a look and then hit the button
